classes/CartRule.php

```diff
@@ -132,6 +134,10 @@ public function add($autodate = true, $null_values = false)
 	public function update($null_values = false)
 	{
 		Cache::clean('getContextualValue_'.$this->id.'_*');
+		
+		if (!$this->reduction_currency)
+			$this->reduction_currency = Configuration::get('PS_CURRENCY_DEFAULT'))->id;
```

Comment:
Oh yeah.. what the hell I wrote here... !

I know now what you want to said, @jeromenadaud !

I try to see what I wrote that... And you knpow why ? It's maybe this line that I follow: https://github.com/PrestaShop/PrestaShop/blob/1.6/classes/controller/FrontController.php#L892

I do a correct now !